### PR TITLE
Improvements to the status subresources

### PR DESCRIPTION
**Pull request by grs**
This changes the CRDs to use conditions instead of simple boolean flags in the status subresources. Various different custom conditions are defined:

Configured - any relevant resources have been updated through the kube API server
Resolved - relevant hosts are resolved if relevant (e.g. for site with linkAccess defined, the site status has the endpoints through which routers can link in)
Running - currently only used for sites to indicate at least one router pod is running
Matched - used for connectors & listeners to indicate whether any matching binding is present (e.g. for a connector, whether there is at least one listener in the network with the same routingKey)
Processed - used for accessgrants to indicate the grant has been processed and verified
Redeemed - used for accesstokens to indicate they have been redeemed    

The standard ready condition is also used. This is generally set when other dependent conditions are met. E.g. a site is Ready when it is Configured, Resolved (if relevant) and Running. Connectors/Listeners are Ready when they are Configured and Matched.

The number of matching connectors is visible for listeners and the number of matching listeners is likewise visible for connectors. The connector status will also show the selected pods where a selector is in use.

### CODE REVIEW

**Review by nluaces — APPROVED** [review]

**Review by lynnemorrison — APPROVED** [review]
code looks great!!   I pulled repo and tested with CLI site, connector, listener commands and was able to bring up link and run hello world example.    I verified the new matchingListenerCounts and matchingConnectorCounts.

One issue I noticed was:
kubectl get routeraccess -n east
No resources found in east namespace.
obviously skupper-router existed and worked fine.

File: pkg/kube/claims/grants.go [hunk 1]
```diff
@@ -220,11 +222,15 @@ func (g *Grants) checkGrant(key string, grant *skupperv1alpha1.AccessGrant) erro
 			changed = true
 		}
 	}
+	var err error
 
 	if len(status) != 0 {
-		grant.Status.StatusMessage = strings.Join(status, ", ")
+		err = fmt.Errorf(strings.Join(status, ", "))
+	}
+	if grant.SetProcessed(err) {
```
[the remark this answers is not in the document]
Comment (grs) [reply pkg/kube/claims/grants.go:230]: The grant is 'Processed' when it has been verified and the code and expiration have been set in the status. It is 'Resolved' when the url and CA are set in the status. These may not happen at the same time. Then when it is both 'Processed' and 'Resolved' it becomes 'Ready'.

**Comment by grs** [comment]
> One issue I noticed was: kubectl get routeraccess -n east No resources found in east namespace. obviously skupper-router existed and worked fine.

did the site in east have linkAccess set? The routeraccess resource is only created on sites that allow incoming link access. Sites that don't enable this will still have a router, but no routeraccess.

**Comment by lynnemorrison** [comment]
> > One issue I noticed was: kubectl get routeraccess -n east No resources found in east namespace. obviously skupper-router existed and worked fine.
> 
> did the site in east have linkAccess set? The routeraccess resource is only created on sites that allow incoming link access. Sites that don't enable this will still have a router, but no routeraccess.

No East side was not configured with linkAccess.  Thank you for the clearification.

**Pull request merged**